monai/transforms/spatial/array.py

```diff
@@ -499,6 +499,13 @@ def __call__(
             warnings.warn("`data_array` is not of type MetaTensor, assuming affine to be identity.")
             # default to identity
             input_affine = np.eye(sr + 1, dtype=np.float64)
+        input_affine_shape = input_affine.shape[0]
+        if input_affine_shape != sr + 1:
```

Comment:
I feel we can compare the length of `self.pixdim` and `sr+1`; otherwise, there might always be a WARNING for 2D data. Also, users may not modify the `affine_matrix`, perhaps we should just mention it in the warning message for users who are unfamiliar with MONAI and don't know they need to add a channel.
What do you think?